cytosnake/tests/functional/test_cli.py

```diff
@@ -0,0 +1,216 @@
+"""
+module: test_cli.py
+
+This testing module composes of functional tests that contains checks for both positive
+negative cases when using CytoSnake's CLI
+
+A positive case indicates that given the user parameters we expect it to run
+successfully.
+
+A negative case indicates that given with the user parameters, our tests are able to
+capture the errors.
+
+Ultimately, test_cli.py will contains functional test to all modes that CytoSnake
+contains.
+"""
+
+import os
+import pathlib
+import shutil
+import subprocess
+from typing import Optional
+
+# import tempfile
+# import pytest
+# import subprocess
+# import test_functions # This will contains helper functions for testing
+from cytosnake.common import errors
+
+
+# -----------------
+# Helper functions
+# -----------------
+class CleanUpHandler:
+    """Used to clean up directories in every single test run"""
+
+    def __init__(self, tmp_path):
+        self.tmp_path = tmp_path
+
+    def __call__(self) -> None:
+        shutil.rmtree(self.tmp_path)
+
+
+def transfer_data(
+    test_dir: pathlib.Path,
+    n_plates: int,
+    n_platemaps: int,
+    metadata_dir_name: Optional[str] = "metadata",
+    testing_data_dir="dummyfiles",
+) -> None:
+    """Wrapper function that transfer datasets found within the pytest module and
+    transfers it to the assigned directory where pytest is conducting the functional
+    tests.
+
+    Parameters
+    ----------
+    test_dir : LocalPath
+        `PyTest.LocalPath` object that contains the path were the test is being
+        conducted
+
+    Return
+    ------
+    None
+        Transfers datafiles from the PyTest module to the testing directory
+    """
+
+    # get files to transfer
+    dataset_dir = pathlib.Path(f"./datasets/{testing_data_dir}").resolve(strict=True)
```

Comment:
Pytest is sometimes used to run the entirety of the "tests" directory without being located within the relative structure of sub-folders. That said, consider making sure this line (and possibly others) are able to run without changes or document the expectations of this test file here (or alternatively within the docstring for the module).

Reply:
Ah I see. Thanks for pointing this out. I doubled checked and added some documentation in regards to this!